subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/IvyPluginPublishingPlugin.java

```diff
@@ -34,9 +34,16 @@
 import org.w3c.dom.Element;
 import org.w3c.dom.Node;
 
+import javax.inject.Inject;
+
 import static org.gradle.plugin.use.resolve.internal.ArtifactRepositoriesPluginResolver.PLUGIN_MARKER_SUFFIX;
 
 abstract class IvyPluginPublishingPlugin implements Plugin<Project> {
+
+    @Inject
+    public IvyPluginPublishingPlugin() {
+        
+    }
```

Comment:
Why is this needed if we do not inject anything?

Also, whitespace is a bit weird now.

Reply:
Interesting, I didn't know about that we don't see default constructors of package-private classes, but do see them if they are declared `public`. I think this would be good to add as a comment here for future visitors of this code.